src/circuit/circuit_frame.rs

```diff
@@ -1250,6 +1267,130 @@ fn reduce_sym<F: LurkField, CS: ConstraintSystem<F>>(
     Ok((output_expr, output_env, output_cont, apply_cont_num))
 }
 
+#[derive(Clone, Copy)]
+pub struct Pointers<F: LurkField> {
+    t: ScalarPtr<F>,
+    nil: ScalarPtr<F>,
+    lambda: ScalarPtr<F>,
+    quote: ScalarPtr<F>,
+    let_: ScalarPtr<F>,
+    letrec: ScalarPtr<F>,
+    cons: ScalarPtr<F>,
+    strcons: ScalarPtr<F>,
+    begin: ScalarPtr<F>,
+    car: ScalarPtr<F>,
+    cdr: ScalarPtr<F>,
+    atom: ScalarPtr<F>,
+    emit: ScalarPtr<F>,
+    sum: ScalarPtr<F>,
+    diff: ScalarPtr<F>,
+    product: ScalarPtr<F>,
+    quotient: ScalarPtr<F>,
+    modulo: ScalarPtr<F>,
+    num_equal: ScalarPtr<F>,
+    equal: ScalarPtr<F>,
+    less: ScalarPtr<F>,
+    less_equal: ScalarPtr<F>,
+    greater: ScalarPtr<F>,
+    greater_equal: ScalarPtr<F>,
+    current_env: ScalarPtr<F>,
+    if_: ScalarPtr<F>,
+    hide: ScalarPtr<F>,
+    commit: ScalarPtr<F>,
+    num: ScalarPtr<F>,
+    u64_: ScalarPtr<F>,
+    comm: ScalarPtr<F>,
+    char_: ScalarPtr<F>,
+    eval: ScalarPtr<F>,
+    open: ScalarPtr<F>,
+    secret: ScalarPtr<F>,
+}
+
+impl<F: LurkField> Pointers<F> {
+    pub fn new(store: &Store<F>) -> Self {
```

Comment:
I refactored all this, moving `Pointers` into `Store`, renaming to `Constants`, and using `OnceCell`. We need one set of constants per store rather than a global static (for example) because the store could theoretically (and probably will sometimes, in the future) be parameterized in such a way that some constant-per-store values are not globally constant. Eventually, it may make sense to move these out of the `Store` also, but for now this strikes a balance of being reasonably manageable and correct, if a little lumpy.